### ISSUE TITLE
DefaultTemplateProposalProvider.getImage() leaks images

### ISSUE THREAD

**Issue by iloveeclipse** [eclipse/xtext-eclipse#1622]
In our tests started with 4.19 / extra VM flag:
`-Dorg.eclipse.swt.graphics.Resource.reportNonDisposed=true`
I see few errors reported with following stack:
```
	at org.eclipse.swt.graphics.Resource.<init>(Resource.java:111)
	at org.eclipse.swt.graphics.Resource.<init>(Resource.java:116)
	at org.eclipse.swt.graphics.Image.<init>(Image.java:653)
	at org.eclipse.jface.resource.URLImageDescriptor.createImage(URLImageDescriptor.java:274)
	at org.eclipse.jface.resource.ImageDescriptor.createImage(ImageDescriptor.java:287)
	at org.eclipse.jface.resource.ImageDescriptor.createImage(ImageDescriptor.java:265)
	at org.eclipse.xtext.ui.editor.templates.DefaultTemplateProposalProvider.getImage(DefaultTemplateProposalProvider.java:87)
```
There is no attempt of either caching or re-using or disposing images created inside `DefaultTemplateProposalProvider.getImage`.

**Comment by cdietrich** [comment]
besides fixing the actual problem we also should run our eclipse and xtend tests with "latest" and `-Dorg.eclipse.swt.graphics.Resource.reportNonDisposed=stacks` and then check the log

**Comment by cdietrich** [comment]
the parameter is actually

`-Dorg.eclipse.swt.graphics.Resource.reportNonDisposed=true`

**Comment by cdietrich** [comment]
@iloveeclipse do i need a special application to reproduce this. with the default i cant. (i can track other non disposes, but not this one)

**Comment by iloveeclipse** [comment]
@cdietrich : I've answered on our internal tracker. I have no clue about application itself and why it is doing whatever it does to call this API, I'm just "the messenger".

**Comment by cdietrich** [comment]
this seems also to be a singleton problem.

**Comment by cdietrich** [comment]
Fixed in 2.25
